### ISSUE TITLE
Merging/Combine ESP Options get lost/overwritten 

### ISSUE THREAD

**Issue by mounirmesselmeni** [anymail/django-anymail#338]
Reporting an error? It's helpful to know:

- Anymail version: 10.1
- ESP (Mailgun, SendGrid, etc.): SparkPost // But irrelevant
- Your ANYMAIL settings (change secrets to "redacted") {"SPARKPOST_SEND_DEFAULTS": {"transactional": True}}
- Versions of Django, requests, python: Django 4.2, Python 3.11
- Exact error message and/or stack trace: No Error
- Any other relevant code and settings (e.g., for problems
  sending, your code that sends the message)

The [documentation](https://anymail.dev/en/stable/sending/anymail_additions/#global-send-defaults) says that when using ESP extra those options will be merged if defined in message.esp_extra, but in reality those options get overriden, if I pass in message.esp_extra for example the pool_name, the default transactional will be missing.

This is because inside anymail's combine function, the python dict.update check only for first level.

In other words:

```python
dict1 = {"options": {"transactional": True}}
dict2 = {"options": {"pool_name": "custom_poolname"}}

dict1.update(dict2)
# will be -> {"options": {"pool_name": "custom_poolname"}}
```

this is what is done in the combine function https://github.com/anymail/django-anymail/blob/main/anymail/utils.py#L50

**Comment by medmunds** [comment]
@mounirmesselmeni thanks for the bug report. We should use a deep merge for esp_extra. (This will only affect ESPs that use nested data structures in their payloads: Sparkpost, Mandrill, maybe a few others I'm forgetting.)

**Comment by medmunds** [comment]
Fix released in Anymail 10.2
